pkg/manager/leaderelection.go

```diff
@@ -2,86 +2,89 @@ package manager
 
 import (
 	"context"
-	"time"
+	"github.com/pkg/errors"
 
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/leaderelection"
-	manager_leaderelection "sigs.k8s.io/controller-runtime/pkg/leaderelection"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/k8snetworkplumbingwg/kubemacpool/pkg/names"
+	poolmanager "github.com/k8snetworkplumbingwg/kubemacpool/pkg/pool-manager"
 )
 
-func (k *KubeMacPoolManager) newLeaderElection(config *rest.Config, scheme *runtime.Scheme) error {
-	recorderProvider, err := NewProvider(config, scheme, log.WithName("events"))
+func (k *KubeMacPoolManager) waitToStartLeading(poolManger *poolmanager.PoolManager) error {
+	<-k.runtimeManager.Elected()
+	// If we reach here then we are in the elected pod.
```

Comment:
I think you have to do poolManager.Start here, so pool manager does not need the mgr.